deb/openresty/debian/openresty.service

```diff
@@ -1,27 +1,28 @@
-# Stop dance for nginx
+# Stop dance for OpenResty
 # =======================
 #
-# ExecStop sends SIGSTOP (graceful stop) to the nginx process.
-# If, after 5s (--retry QUIT/5) nginx is still running, systemd takes control
+# ExecStop sends SIGSTOP (graceful stop) to the OpenResty process.
+# If, after 5s (--retry QUIT/5) OpenResty is still running, systemd takes control
 # and sends SIGTERM (fast shutdown) to the main process.
-# After another 5s (TimeoutStopSec=5), and if nginx is alive, systemd sends
+# After another 5s (TimeoutStopSec=5), and if OpenResty is alive, systemd sends
 # SIGKILL to all the remaining processes in the process group (KillMode=mixed).
 #
 # nginx signals reference doc:
 # http://nginx.org/en/docs/control.html
 #
 [Unit]
-Description=full-fledged web platform
+Description=OpenResty (full-fledged web platform)
 After=network.target
 
 [Service]
+#This is Default Prefix build in OpenResty official repo
+Environment=OPENRESTY_PREFIX=/usr/local/openresty/nginx/
+#User can overwrite this environment via `systemctl edit openresty` to create a override file
+#In the world of systemd, PIDFile is not very important, $MAINPID will be set to the main process of the daemon
 Type=forking
-PIDFile=/usr/local/openresty/nginx/logs/nginx.pid
-ExecStartPre=/usr/local/openresty/nginx/sbin/nginx -t -q -g 'daemon on; master_process on;'
-ExecStart=/usr/local/openresty/nginx/sbin/nginx -g 'daemon on; master_process on;'
-ExecReload=/usr/local/openresty/nginx/sbin/nginx -g 'daemon on; master_process on;' -s reload
-ExecStop=-/sbin/start-stop-daemon --quiet --stop --retry QUIT/5 --pidfile /usr/local/openresty/nginx/logs/nginx.pid
```

Comment:
Maybe a replacement of $MAINPID to /usr/local/openresty/nginx/logs/nginx.pid is enough